client-admin/gulpfile.js

```diff
@@ -1,503 +0,0 @@
-  var indexDest = [destRootBase, "index_admin.html"].join("/");
-  // fs.mkdirSync(destRootBase);
-  fs.writeFileSync(indexDest, html);
-});
-
-gulp.task("embed", [], function () {
-  var index = fs.readFileSync("embed.html", { encoding: "utf8" });
-  var dest = [destRootBase, "embed.html"].join("/");
-  // fs.mkdirSync(destRootBase);
-  fs.writeFileSync(dest, index);
-});
-
-gulp.task("embedPreprod", [], function () {
-  var index = fs.readFileSync("embedPreprod.html", { encoding: "utf8" });
-  var dest = [destRootBase, "embedPreprod.html"].join("/");
-  // fs.mkdirSync(destRootBase);
-  fs.writeFileSync(dest, index);
-});
-
-gulp.task("embedReport", [], function () {
-  var index = fs.readFileSync("embedReport.html", { encoding: "utf8" });
-  var dest = [destRootBase, "embedReport.html"].join("/");
-  // fs.mkdirSync(destRootBase);
-  fs.writeFileSync(dest, index);
-});
-
-gulp.task("embedReportPreprod", [], function () {
-  var index = fs.readFileSync("embedReportPreprod.html", { encoding: "utf8" });
-  var dest = [destRootBase, "embedReportPreprod.html"].join("/");
-  // fs.mkdirSync(destRootBase);
-  fs.writeFileSync(dest, index);
-});
-
-gulp.task("404", [], function () {
-  var index = fs.readFileSync("404.html", { encoding: "utf8" });
-  var dest = [destRootBase, "404.html"].join("/");
-  // fs.mkdirSync(destRootBase);
-  fs.writeFileSync(dest, index);
-});
-
-gulp.task("preprodConfig", function () {
-  preprodMode = true;
-  minified = true;
-  scpSubdir = polisConfig.SCP_SUBDIR_PREPROD;
-  s3Subdir = polisConfig.S3_BUCKET_PREPROD;
-});
-
-gulp.task("unminifiedConfig", function () {
-  minified = false;
-});
-
-gulp.task("prodConfig", function () {
-  prodMode = true;
-  minified = true;
-  scpSubdir = polisConfig.SCP_SUBDIR_PROD;
-  s3Subdir = polisConfig.S3_BUCKET_PROD;
-});
-
-gulp.task("configureForProduction", function (callback) {
-  devMode = false;
-  destRootBase = "dist";
-
-  console.log("getGitHash begin");
-  // NOTE using callback instead of returning a promise since the promise isn't doing the trick - haven't tried updating gulp yet.
-  getGitHash()
-    .then(function (hash) {
-      var d = new Date();
-      var tokenParts = [
-        d.getFullYear(),
-        d.getMonth() + 1,
-        d.getDate(),
-        d.getHours(),
-        d.getMinutes(),
-        d.getSeconds(),
-      ];
-
-      if (hash) {
-        hash = hash.toString().match(/[A-Za-z0-9]+/)[0];
-        tokenParts.push(hash);
-      }
-
-      var unique_token = tokenParts.join("_");
-      destRootRest = [staticFilesPrefix, unique_token].join("/");
-      versionString = unique_token;
-
-      console.log(
-        "done setting destRoot: " +
-          destRoot() +
-          "  destRootRest: " +
-          destRootRest +
-          "  destRootBase: " +
-          destRootBase
-      );
-      callback(null);
-    })
-    .catch(function (err) {
-      console.error("getGitHash err");
-      console.error(err);
-      callback(true);
-    });
-});
```

Comment:
We actually needed this part of the routine for caching to work properly

Reply:
Webpack takes care of 'cache busting' for you for the Javascript import - in Webpack config line 32 it uses a magic string '[chunkhash:8]' to generate a random 8 character string and puts that in the output filename. In the `HtmlWebpackPlugin` it is injected into the HTML body when it is processed - some earlier discussion goes intot this at https://github.com/compdemocracy/polis/pull/1242/files#r776461657